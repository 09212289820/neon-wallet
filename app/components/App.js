import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import NetworkSwitch from './NetworkSwitch';

// import { Link } from 'react-router';

let App = ({ children }) => {
  return <div id="pageWrapper">{ children }<NetworkSwitch /></div>;
};

const mapStateToProps = (state) => ({
});

App = connect(mapStateToProps)(App);

export default App;
