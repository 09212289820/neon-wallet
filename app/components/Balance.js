import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router'
import { getBalance } from '../wallet/api.js';
import { setBalance } from '../actions/index.js';

const initiateGetBalance = (dispatch, address) => {
  getBalance(address).then(function(result){
    dispatch(setBalance(result.ANS, result.ANC));
  })
};

class Balance extends Component {

  componentDidMount = () => {
    initiateGetBalance(this.props.dispatch, this.props.address)
  }

  render = () =>
    <div id="balancePage">
    <div className="title">Account Balance:</div>
      <div id="publicAddress"><span className="descriptor">Address:</span><span className="key">{ this.props.address }</span></div>
      <div id="balanceList">
        <div><span className="asset">AntShares:</span><span className="amount">{ this.props.ans }</span></div>
        <div><span className="asset">AntCoins:</span><span className="amount">{ this.props.anc }</span></div>
      </div>
      <div className="margin10"><button><Link to="/send">Send Transaction</Link></button></div>
      <button><Link to="/">Logout</Link></button>
    </div>;
}

const mapStateToProps = (state) => ({
  ans: state.wallet.ANS,
  anc: state.wallet.ANC,
  address: state.account.address,
});

Balance = connect(mapStateToProps)(Balance);

export default Balance;
