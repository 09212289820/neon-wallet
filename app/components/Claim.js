import React, { Component } from 'react';
import { connect } from 'react-redux';
import { doClaimAllGas, doSendAsset } from '../wallet/api.js';
import { sendEvent, clearTransactionEvent, setClaimRequest, disableClaim } from '../actions/index.js';
import ReactTooltip from 'react-tooltip'

const doGasClaim = (dispatch, net, wif, selfAddress, ans) => {
  dispatch(sendEvent(true, "Sending Neo to Yourself..."));
  doSendAsset(net, selfAddress, wif, "Neo", ans).then((response) => {
    if (response.result === undefined){
      dispatch(sendEvent(false, "Transaction failed!"));
    } else {
      dispatch(sendEvent(true, "Waiting for transaction to clear..."));
      dispatch(setClaimRequest(true));
      dispatch(disableClaim(true));
    }
  });
}

class Claim extends Component {

  componentDidUpdate = () => {
    console.log(this.props);
    if (this.props.claimRequest === true && this.props.claimWasUpdated == true){
      this.props.dispatch(setClaimRequest(false));
      doClaimAllGas(this.props.net, this.props.wif).then((response) => {
        if (response.result === true){
          this.props.dispatch(sendEvent(true, "Claim was successful! Your balance will update once the blockchain has processed it."));
          setTimeout(() => this.props.dispatch(disableClaim(false)), 60000);
        } else {
          this.props.dispatch(sendEvent(false, "Claim failed"))
        }
        setTimeout(() => this.props.dispatch(clearTransactionEvent()), 5000);
      });
    }
  }

  render = () => {
    let renderButton;
    const doClaim = () => doGasClaim(this.props.dispatch, this.props.net, this.props.wif, this.props.address, this.props.ans);
    if (this.props.disableClaimButton === false){
      renderButton = <button onClick={doClaim}>Claim {this.props.claimAmount} GAS</button>;
    } else {
      renderButton = (<div>
          <button data-tip data-for="claimTip" className="disabled">Claim {this.props.claimAmount} GAS</button>
          <ReactTooltip class="solidTip" id="claimTip" place="bottom" type="dark" effect="solid">
            <span>You can claim Gas once every 5 minutes</span>
          </ReactTooltip>
        </div>);
    }
    return <div id="claim">{renderButton}</div>;
  }

}

const mapStateToProps = (state) => ({
  claimAmount: state.wallet.claimAmount,
  claimRequest: state.wallet.claimRequest,
  claimWasUpdated: state.wallet.claimWasUpdated,
  disableClaimButton: state.wallet.disableClaimButton,
  wif: state.account.wif,
  address: state.account.address,
  net: state.wallet.net,
  ans: state.wallet.ANS
});

Claim = connect(mapStateToProps)(Claim);

export default Claim;
