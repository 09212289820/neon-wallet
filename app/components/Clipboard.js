import { clipboard } from 'electron';

export const writeText = (content) => {
  clipboard.writeText(content, 'selection');
};