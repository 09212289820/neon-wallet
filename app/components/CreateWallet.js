import React, { Component } from 'react';
import { connect } from 'react-redux';
import { newWallet } from '../actions/index.js';
import { Link } from 'react-router';
import WalletInfo from './WalletInfo.js'

const generateWallet = (dispatch) => {
  dispatch(newWallet());
};

class CreateWallet extends Component {

  componentDidMount = () => {
    generateWallet(this.props.dispatch);
  }

  render = () =>
    <div id="newWallet">
      <div className="title">Generate a new wallet:</div>
      <WalletInfo wif={this.props.wif} address={this.props.address}/>
      <button onClick={() => generateWallet(this.props.dispatch)}>Regenerate wallet</button>
      <button><Link to="/">Back to Login</Link></button>
    </div>;

}

const mapStateToProps = (state) => ({
  wif: state.generateWallet.wif,
  address: state.generateWallet.address
});

CreateWallet = connect(mapStateToProps)(CreateWallet);

export default CreateWallet;
