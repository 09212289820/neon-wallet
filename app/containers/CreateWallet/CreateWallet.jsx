// @flow
import React, { Component } from 'react'
import { generateEncryptedWif } from 'neon-js'
import { isNil } from 'lodash'
import { Link } from 'react-router'
import DisplayWalletKeys from '../../components/DisplayWalletKeys'
import { validatePassphrase } from '../../core/wallet'
import Logo from '../../components/Logo'

type Props = {
    newWallet: Function,
    sendEvent: Function,
    clearTransactionEvent: Function,
    resetKey: Function,
    address: string,
    generating: boolean,
    wif: string,
    encryptedWif: string,
    passphrase: string,
}

type State = {
  passphrase: string,
  passphrase2: string,
}

export default class CreateWallet extends Component<Props, State> {
  state = {
    passphrase: '',
    passphrase2: ''
  }

  generateNewWallet = () => {
    const { passphrase, passphrase2 } = this.state
    const { clearTransactionEvent, sendEvent, newWallet } = this.props

    if (!passphrase || !passphrase2) return null

    if (passphrase !== passphrase2) {
      sendEvent(false, 'Passphrases do not match')
      setTimeout(() => clearTransactionEvent(), 5000)
      return
    }
    if (validatePassphrase(passphrase)) {
      // TODO: for some reason this blocks, so giving time to processes the earlier
      // dispatch to display "generating" text, should fix this in future
      sendEvent(true, 'Generating encoded key...')
      setTimeout(() => {
        generateEncryptedWif(passphrase).then((result) => {
          newWallet(result)
          clearTransactionEvent()
        })
      }, 500)
    } else {
      sendEvent(false, 'Please choose a longer passphrase')
      setTimeout(() => clearTransactionEvent(), 5000)
      this.resetPassphrases()
    }
  }

  resetPassphrases () {
    this.setState({
      passphrase: '',
      passphrase2: ''
    })
  }

  render () {
    const { generating, address, encryptedWif, resetKey, clearTransactionEvent, sendEvent } = this.props
    const { passphrase, passphrase2 } = this.state

    const passphraseDiv = (
      <div>
        <div className='info'>
          Choose a passphrase to encrypt your private key:
        </div>
        <input
          type='text'
          placeholder='Enter passphrase here'
          value={passphrase}
          onChange={(e) => this.setState({ passphrase: e.target.value })}
        />
        <input
          type='text'
          placeholder='Repeat passphrase here'
          value={passphrase2}
          onChange={(e) => this.setState({ passphrase2: e.target.value })}
        />
        <button onClick={this.generateNewWallet} > Generate keys </button>
        <Link to='/'><button className='altButton'>Home</button></Link>
      </div>
    )

    return (
      <div id='newWallet'>
        <Logo />
        {isNil(this.props.wif) ? passphraseDiv : <div />}
        {generating && <div className='generating'>Generating keys...</div>}
        {!generating && !isNil(this.props.wif) &&
        <DisplayWalletKeys
          address={address}
          wif={this.props.wif}
          passphrase={this.props.passphrase}
          passphraseKey={encryptedWif}
          resetKey={resetKey}
          sendEvent={sendEvent}
          clearTransactionEvent={clearTransactionEvent}
        />
        }
      </div>
    )
  }
}
