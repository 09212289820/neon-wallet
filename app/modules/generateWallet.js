import { getAccountsFromWIFKey, generatePrivateKey, getWIFFromPrivateKey } from 'neon-js';
import { encrypt_wif, decrypt_wif } from '../util/Passphrase.js';

// Constants
const NEW_WALLET_KEYS = 'NEW_WALLET_KEYS';
const NEW_WALLET = 'NEW_WALLET';


export function newWalletKeys(passphrase){
  return {
    type: NEW_WALLET_KEYS,
    passphrase
  }
}

export function newWallet(account){
  return {
    type: NEW_WALLET,
    wif: account.wif,
    address: account.address,
    passphrase: account.passphrase,
    encryptedWif: account.encryptedWif
  }
}

// Reducer used for state necessary to generating a wallet
export default (state = {wif: null, address:null, passphrase: null, encryptedWif: null}, action) => {
  switch (action.type) {
    case NEW_WALLET_KEYS:
      const newPrivateKey = generatePrivateKey();
      const newWif = getWIFFromPrivateKey(newPrivateKey);
      const encryptedWif = encrypt_wif(newWif, action.passphrase);
      const loadAccount = getAccountsFromWIFKey(newWif);
      return {...state, wif:newWif, address:loadAccount[0].address, passphrase: action.passphrase, encryptedWif: encryptedWif};
    case NEW_WALLET:
      return  {...state, wif: action.wif, address: action.address, passphrase: action.passphrase, encryptedWif: action.encryptedWif};
    default:
      return state;
  }
};
