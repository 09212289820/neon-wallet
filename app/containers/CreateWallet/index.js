// @flow
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { newWallet, resetKey } from '../../modules/generateWallet'
import { sendEvent, clearTransactionEvent } from '../../modules/transactions'
import CreateWallet from './CreateWallet'

const mapStateToProps = (state) => ({
  wif: state.generateWallet.wif,
  address: state.generateWallet.address,
  encryptedWif: state.generateWallet.encryptedWif,
  passphrase: state.generateWallet.passphrase,
  generating: state.generateWallet.generating
})

const actionCreators = {
  newWallet,
  sendEvent,
  clearTransactionEvent,
  resetKey
}

const mapDispatchToProps = dispatch => bindActionCreators(actionCreators, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(CreateWallet)
