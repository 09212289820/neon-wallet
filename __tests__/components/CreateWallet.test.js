import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import QRCode from 'qrcode';
import CreateWallet from '../../app/components/CreateWallet';
import * as Clipboard from '../../app/components/Clipboard';
import { newWallet } from '../../app/modules/generateWallet';

const state = {
  generateWallet: {
    wif: 'wif',
    address: 'address'
  }
}

describe('CreateWallet', () => {
  test('should render without crashing', () => {
    const store = configureStore()(state);
    const wrapper = shallow(<CreateWallet store={store} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });
  
  test('should dispatch new wallet action', () => {
    const store = configureStore()(state);
    const wrapper = mount(<CreateWallet store={store} />);
    expect(store.getActions()[0]).toEqual(newWallet());
  });
  
  test('should dispatch new wallet action', () => {
    const store = configureStore()(state);
    const wrapper = mount(<CreateWallet store={store} />);
    const toCanvas = QRCode.toCanvas = jest.fn();
    wrapper.setProps({ target: '' });
    expect(toCanvas.mock.calls.length).toEqual(2);
    expect(toCanvas).toHaveBeenCalledWith(expect.any(Object), 'address', {version: 5}, expect.any(Function));
    expect(toCanvas).toHaveBeenCalledWith(expect.any(Object), 'wif', {version: 5}, expect.any(Function));
  });

  test('should copy address to clipboard when clicked', () => {
    const store = configureStore()(state);
    const wrapper = shallow(<CreateWallet store={store} />);
    Clipboard.writeText = jest.fn();
    wrapper.dive().find('.copyKey').at(0).simulate('click');
    expect(Clipboard.writeText).toHaveBeenCalledWith('address');
  });

  test('should copy wif to clipboard when clicked', () => {
    const store = configureStore()(state);
    const wrapper = shallow(<CreateWallet store={store} />);
    Clipboard.writeText = jest.fn();
    wrapper.dive().find('.copyKey').at(1).simulate('click');
    expect(Clipboard.writeText).toHaveBeenCalledWith('wif');
  });
});