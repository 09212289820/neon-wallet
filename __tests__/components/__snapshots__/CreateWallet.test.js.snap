// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWallet should render without crashing 1`] = `
<div
  id="newWallet"
>
  <div
    className="disclaimer"
  >
    Save the keys below. We will never show you this private key again.
  </div>
  <div
    className="addressBox"
  >
    <canvas />
    <div>
      Public Address
    </div>
  </div>
  <div
    className="privateKeyBox"
  >
    <canvas />
    <div>
      Private Key (WIF)
    </div>
  </div>
  <div
    className="keyList"
  >
    <span
      className="label"
    >
      Public Address:
    </span>
    <span
      className="key"
    >
      address
    </span>
    <span
      className="copyKey"
      onClick={[Function]}
    >
      <MdContentCopy
        data-for="copyPublicKeyTip"
        data-tip={true}
      />
    </span>
  </div>
  <div
    className="keyList"
  >
    <span
      className="label"
    >
      Private Key:
    </span>
    <span
      className="key"
    >
      wif
    </span>
    <span
      className="copyKey"
      onClick={[Function]}
    >
      <MdContentCopy
        data-for="copyPrivateKeyTip"
        data-tip={true}
      />
    </span>
  </div>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/"
  >
    <button>
      Back to Login
    </button>
  </Link>
  <ReactTooltip
    class="solidTip"
    effect="solid"
    id="copyPublicKeyTip"
    insecure={true}
    place="bottom"
    resizeHide={true}
    type="dark"
    wrapper="div"
  >
    <span>
      Copy Public Key
    </span>
  </ReactTooltip>
  <ReactTooltip
    class="solidTip"
    effect="solid"
    id="copyPrivateKeyTip"
    insecure={true}
    place="bottom"
    resizeHide={true}
    type="dark"
    wrapper="div"
  >
    <span>
      Copy Private Key
    </span>
  </ReactTooltip>
</div>
`;
